### PR TITLE
Use apt preferences to manage dependency version mismatches

### PR DESCRIPTION
**Pull request by amcmanus**
In some cases, it's possible for apt to update a package but leave the older version available in the latest main + security repo. When package dependencies are also updated in this manner, the dependency versions can result in a mismatch that will not allow the package to be installed. 

This PR provides a mechanism to use apt preferences to specify the repo for packages experiencing this issue. When this dependency mismatch occurs, we ensure the affected packages use the snapshot repository, even if they may also be available in the latest main still. In most cases, this problem will resolve itself once the older packages are removed, but this provides a solution in the case a build fails with this issue.

### CODE REVIEW

File: inventories/v3.1/group_vars/all/main.yaml [hunk 1]
```diff
@@ -12,4 +12,5 @@ vm_preseed_file: "production-preseed.cfg"
 secure_boot: true
 rust_version: "1.72.0"
 cloned_images:
-  - vx
+  - online
+  - offline
```
Comment (benadida) [inline inventories/v3.1/group_vars/all/main.yaml:16]: is this an unrelated change?

**Review by benadida — APPROVED** [review]

File: inventories/v3.1/group_vars/all/main.yaml (hunk 1 above)
Comment (amcmanus) [reply inventories/v3.1/group_vars/all/main.yaml:16]: It's unrelated to the apt bit, but it's necessary for the full trusted build to work with this inventory since we need separate images, so I went ahead and included it.
